### PR TITLE
[MNG-8340] Resolve parent according to the exact model location

### PR DESCRIPTION
**Pull request by gnodet**
JIRA issue: [MNG-8340](https://issues.apache.org/jira/browse/MNG-8340)

We need to double check one last time that it works, or better, add an IT.

### CODE REVIEW

**Comment by cstamas** [comment]
Just a plain project with one module shade + dep reduced POM would work as an IT?

**Review by cstamas — APPROVED** [review]

**Comment by gnodet** [comment]
> Just a plain project with one module shade + dep reduced POM would work as an IT?

It needs child subproject, and the shade plugin should generate the Pom in the target folder (instead of in the same directory than the Pom).

**Comment by cstamas** [comment]
Something is still not right, as this PR w/ shade in IT https://github.com/apache/maven-integration-testing/pull/400 now causes NPE

**Comment by gnodet** [comment]
It seems this PR also fixes [MNG-8336](https://issues.apache.org/jira/browse/MNG-8336)

File: impl/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java [hunk 1]
```diff
@@ -913,7 +913,7 @@ private Model injectLifecycleBindings(
         }
         project.setPluginArtifactRepositories(pluginRepositories);
 
-        if (request.getRequestType() == ModelBuilderRequest.RequestType.BUILD_POM) {
+        if (request.getRequestType() == ModelBuilderRequest.RequestType.BUILD_PROJECT) {
```
Comment (cstamas) [inline impl/maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java:916]: One correct change from my "guesswork" :smile:

**Review by cstamas — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #9515

**Pull request merged**